metadata.go

```diff
@@ -192,6 +200,86 @@ func GetImageToMDMap(imageMDs []ImageMetadataInfo) (imageToMDMap map[string][]Im
 	return
 }
 
+// getLocalImages queries the local Docker daemon for list of images.
+func getLocalImages() (imageMap map[ImageIDType][]RepoTagType, e error) {
+
+	// query a list of images from Docker daemon
+	response, e := listImages()
+	if e != nil {
+		return nil, e
+	}
+	blog.Info(string(response))
+	// parse JSON
+	var localImageList []LocalImageStruct
+	if e = json.Unmarshal(response, &localImageList); e != nil {
+		return nil, e
+	}
+
+	// make map from each imageID to all of its aliases (repo+tag)
+	imageMap = make(map[ImageIDType][]RepoTagType)
+	for _, localImage := range localImageList {
+		imageID := ImageIDType(localImage.ID)
+		for _, repoTag := range localImage.RepoTags {
+			// skip images with no repo:tag
+			if repoTag == "" {
+				blog.Info("Image ", imageID, " has a repoTag of empty string.")
+				continue
+			}
+			// repoTag example: "localhost:5000/test/busybox:latest"
+			// repo: "localhost:5000/test/busybox"
+			// tag: "latest"
+			ss := strings.Split(repoTag, ":")
+			if len(ss) < 2 {
```

Comment:
A repoTag may contain more than one ':' delimiters, for example in
"localhost:5000/test/busybox:latest".

We need to take the last sub-string 'latest' as a tag and the
remainder including the first ':' as a repo string.


On Mon, Jul 20, 2015 at 10:55 AM, yoshiotu <notifications@github.com> wrote:

> In metadata.go
> <https://github.com/banyanops/collector/pull/19#discussion_r35023291>:
>
> > +
> > +	// make map from each imageID to all of its aliases (repo+tag)
> > +	imageMap = make(map[ImageIDType][]RepoTagType)
> > +	for _, localImage := range localImageList {
> > +		imageID := ImageIDType(localImage.ID)
> > +		for _, repoTag := range localImage.RepoTags {
> > +			// skip images with no repo:tag
> > +			if repoTag == "" {
> > +				blog.Info("Image ", imageID, " has a repoTag of empty string.")
> > +				continue
> > +			}
> > +			// repoTag example: "localhost:5000/test/busybox:latest"
> > +			// repo: "localhost:5000/test/busybox"
> > +			// tag: "latest"
> > +			ss := strings.Split(repoTag, ":")
> > +			if len(ss) < 2 {
>
> Should this test be "if len(ss) != 2" ? Then the following lines could be
> simplified, "tag := repotag[1]" and "repo := repotag[0]"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/banyanops/collector/pull/19/files#r35023291>.
>